### ISSUE TITLE
bike makes emergency braking when meeting pedestrian within a intersection

### ISSUE THREAD

**Issue by yunpangfloetteroed** [eclipse/sumo#7684]
It seems that bike cannot corretly detect pedestrian in best time within a intersection and, therefore, makes emergency braking. Moreover, the corresponding pedestrian did not change his/her "lane" under a sublane model to avoid the possible conflict with the bike. At the same time, bike keeps stoping after emergency braking until the pedestrian passed. (see the attached files with the prefix "test2")

When the same network is cutted and only a small area (only the problem intersection) is left, the above mentioned problem cannot be found. It looks like that the pedestrian used different internal links to pass the intersection. (see the attached files with the prefix "test")

**Comment by yunpangfloetteroed** [comment]
Thanks for the notice! It is wierd that the uploaded zip file is disappeared...  You find attached the zip file.
[test.zip](https://github.com/eclipse/sumo/files/5458018/test.zip)

**Comment by namdre** [comment]
I think the main issue here is that pedestrians don't care which side of the road they walk on when all lanes in both directions are usable (no car-lane in between).
So far most simulations only had a single lane that was used in both directions and they walk on the right within that lane.
 At least in this case they should prefere the right side.

In your two versions of the scenario the pedestian uses different sides of the road when approaching the junction.

**Comment by namdre** [comment]
@behrisch: how about an edge penalty for the pedestrian routing on all edges that allow vehicles when using them against the driving direction. Something along the lines of
`--persontrip.walk-opposite-factor` (default < 1)

**Comment by yunpangfloetteroed** [comment]
Does the choice which side to walk happen randomly in the current version? --> therefore, the ped. uses different sides of the road in these two tests.
In the test2, it looks like that the pedestrian ignore the cyclist and the cyclist stops until the pedestrian passes. Could the ped and the cyclist detect each other eailer before they enter the intersection? Then they could try to choose other connections to aviod the confilct?

**Comment by namdre** [comment]
The connection is chosing earlier when computing the routes. If both sides of the road lead to similar walking distances the route choice is arbitrary (probably based on sorting of ids).

**Comment by yunpangfloetteroed** [comment]
Are there more connections within an intresection when the sublane model is activated? If it is the case, (1) the activation of rerouting device or (2) re-updating connections at intersections (this does not exist in sumo yet) could help to choose a new connection. Right?

**Comment by namdre** [comment]
No. There is only one connection for each lane-to-lane relation.

**Comment by behrisch** [comment]
> @behrisch: how about an edge penalty for the pedestrian routing on all edges that allow vehicles when using them against the driving direction. Something along the lines of
> `--persontrip.walk-opposite-factor` (default < 1)

I am still not sure how this would solve the original issue. The person may still choose to use the edge.

**Comment by yunpangfloetteroed** [comment]
I just increased the number of peds in the test scenario. It looks like more than one lane-to-lane connection exists within the intersection for pedestrains (see the attached figure). or it is only visualization? 
![peds_at_intersection](https://user-images.githubusercontent.com/26454969/97713036-94a74d00-1abf-11eb-9e86-e5839e3edcd8.png)

**Comment by namdre** [comment]
@behrisch You are right. I just think that the original situation would be a lot rarer and improving the edge choice is more relevant than the improving the interaction when switching sides. I moved my suggestion to #7730

Regarding the original issue: Vehicles on a walkingarea are treated like static obstacles by the pedestrians. Instead some space ahead of the vehicle (according to it's speed) should also be blocked. The same concept is already used on normal lanes and crossings.

**Comment by namdre** [comment]
@yunpangfloetteroed There is only one connection but the pedestrians may use different lateral offsets when walking (somewhat similar to the sublane model).

**Comment by yunpangfloetteroed** [comment]
Regarding the original issue: 1. I think it is fine that vehicles on a walkingarea are treated like static obstacles by the pedestrians at/within intersections (they expect that vehicles should "yield" them). Vehicle should be able to detect pedestrians in good time so that they do not need to make emergency braking, especially bikes.

2. When pedestrians always use the rightmost lane in the striping model, we expect then to see pedestrians walk on the right-hand lane/road (in the same direction) after corssing the intesection in the picture shown above. right? But they walk in the opposition direction currently.

**Comment by namdre** [comment]
1) I disagree because there are several paths across a walkingarea and some of them do not conflict with vehicular traffic. With your approach vehicles would slow down in unnecessary situations. While on a shared space the right-of-way rules are more fluid and vehicles as well as pedestrians have to slow down some of the time.

2) That is a misunderstanding. Right now pedestrians walk on the right *within* a lane but the routing algorithm that selects the edge does not care about right or left (which my suggested option would hopefully fix).

**Comment by yunpangfloetteroed** [comment]
After the discussion with Jakob, seperate tickets will be generated for: 
- different lateral offsets will be used by bikes/vehicles within intersections (sublane-behaviour within in intersections)
- pedestrians walk on the rightmost side 'edge' of a road at each direction
- consideration of speed variation/changes for bikes/peds when meeting moving objects (in opposite direction).
@namdre right?

**Comment by namdre** [comment]
I'm not sure about the meaning of the last point but I've added links to releated tickets at the top.

**Comment by yunpangfloetteroed** [comment]
the last one is with regard to that bikes/peds would consider to adjust their speeds regarding the approaching objects, also in opposite direction in a walking area. Do I understand it correctly?

**Comment by namdre** [comment]
some new thoughts on the situation:
- pedestrians should try to create more space to oncoming vehicles even if they are already on distinct stripes
- vehicles should take into account pedestrian angle in order to avoid braking when both trajectories can pass without conflict

**Comment by yunpangfloetteroed** [comment]
Regarding the second point, bikes should still be able to earlier detect possilbe conflicts (something like enlarging the safety gap?) and start earlier to brake for avoiding emergency braking.
